build-mbl/poky-build/poky-bitbake-wrapper.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+# Copyright (c) 2019, Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+set -e
+set -u
+
+WORKAREA="$1"; shift;
+cd "${WORKAREA}/layers/poky"
+
+set +u
+set +e
+# shellcheck disable=SC1091
+TEMPLATECONF=meta-poky/conf source oe-init-build-env
+set -e
+set -u
+
+bitbake-layers add-layer ../../meta-freescale/
```

Comment:
I'm not really keen on having these `add-layer` commands in this script  - are they idempotent?

Instead of having a "bitbake" wrapper, maybe we should have a "bitbake-env" wrapper that just sources `oe-init-build-env` and runs `"$@"` rather than `bitbake "$@"`. Then we could call it three times:
```
poky-bitbake-env-wrapper.sh bitbake-layers add-layer ../../meta-freescale/
poky-bitbake-env-wrapper.sh bitbake-layers add-layer ../../meta-psa/
poky-bitbake-env-wrapper.sh bitbake  "$image"
```
(Well, the Python equivalent of that).

Or we could have a generic "source-then-run" script that takes the script to `source` as the first param and a command to run as the remaining params. Then it could be used for both the Poky build and the Edge build, and potentially the MBL build if we ever translate it to a non-Shell language.

Reply:
Moved the add-layer commands into the Python file and changed the shell script accordingly.

Reply:
Thanks.